rllib/models/specs/checker.py

```diff
@@ -0,0 +1,290 @@
+from collections import abc
+import functools
+from typing import Union, Mapping, Any, Callable
+
+from ray.util.annotations import DeveloperAPI
+
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.models.specs.specs_base import Spec, TypeSpec
+from ray.rllib.models.specs.specs_dict import SpecDict
+from ray.rllib.models.specs.typing import SpecType
+
+
+def _convert_to_canonical_format(spec: SpecType) -> Union[Spec, SpecDict]:
+    """Converts a spec type input to the canonical format.
+
+    The canonical format is either a tree structure with SpecType leaves (i.e.
+    SpecDict) or if the spec is a single constraint, a single SpecType object.
+
+    The input can be any of the following:
+        - a list of nested_keys. nested_keys are either strings or tuples of strings
+        specifying the path to a leaf in the tree.
+        - a tree of constraints. The tree structure can be specified as any nested
+        hash-map structure (e.g. dict, SpecDict, NestedDict, etc.) The leaves of the
+        tree can be either a Spec object, a type, or None. If the leaf is a type, it is
```

Comment:
The code below looks more like "If the leaf is a `type or a tuple`.

Reply:
it's fine. I think the SpecDict will break if it's a tuple of other stuff :). The doc says tuple of strings and that is the natural way of using this data structure.